magda-web-client/src/Components/Common/Pagination.js

```diff
@@ -1,41 +1,56 @@
 import React, { Component } from "react";
-import left_arrow from "assets/left-arrow.svg";
-import right_arrow from "assets/right-arrow.svg";
 import "./Pagination.scss";
 import { needsContent } from "helpers/content";
+import { Link } from "react-router-dom";
+import queryString from "query-string";
 
 class Pagination extends Component {
     constructor(props) {
         super(props);
-        this.onClick = this.onClick.bind(this);
         this.renderNextButton = this.renderNextButton.bind(this);
         this.renderPrevButton = this.renderPrevButton.bind(this);
+        this.generateRoute = this.generateRoute.bind(this);
     }
 
-    onClick(page) {
-        this.props.onPageChange(page);
+    // generates the href for each link-list-item in pagination list.
+    generateRoute(page) {
+        return (
+            `search?` +
```

Comment:
Might want to put a leading `/` in front of this, just in case it ever gets invoked from something like `/search/otherPath?etc` because that'll cause it to go to `/search/search` (I think?).

Reply:
yes good point! I've changed it so it uses `this.props.location.pathname` (which is basically the same as '/search' )